user.Current may return err in some cases on Windows

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x40 pc=0xae5ea2]

goroutine 1 [running]:
github.com/ferama/rospo/cmd.init.1()
        github.com/ferama/rospo/cmd/grabpubkey.go:15 +0x62
```

https://github.com/golang/go/issues/37348

I think it is better to use os.Getwd or empty string as a fallback.

--- a/cmd/cmnflags/sshc.go
+++ b/cmd/cmnflags/sshc.go
@@ -1,10 +1,10 @@
 package cmnflags
 
 import (
-	"os/user"
 	"path/filepath"
 
 	"github.com/ferama/rospo/pkg/sshc"
+	"github.com/ferama/rospo/pkg/utils"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
@@ -12,7 +12,7 @@ import (
 // AddSshClientFlags adds sshc common flags to FlagSet
 func AddSshClientFlags(fs *pflag.FlagSet) {
 
-	usr, _ := user.Current()
+	usr := utils.CurrentUser()
 	defaultIdentity := filepath.Join(usr.HomeDir, ".ssh", "id_rsa")
 	knownHostFile := filepath.Join(usr.HomeDir, ".ssh", "known_hosts")
 

--- a/cmd/grabpubkey.go
+++ b/cmd/grabpubkey.go
@@ -1,17 +1,17 @@
 package cmd
 
 import (
-	"os/user"
 	"path/filepath"
 
 	"github.com/ferama/rospo/pkg/sshc"
+	"github.com/ferama/rospo/pkg/utils"
 	"github.com/spf13/cobra"
 )
 
 func init() {
 	rootCmd.AddCommand(grabpubkeyCmd)
 
-	usr, _ := user.Current()
+	usr := utils.CurrentUser()
 	knownHostFile := filepath.Join(usr.HomeDir, ".ssh", "known_hosts")
 	grabpubkeyCmd.PersistentFlags().StringP("known-hosts", "k", knownHostFile, "the known_hosts file absolute path")
 }

--- a/pkg/sshc/ssh_connection.go
+++ b/pkg/sshc/ssh_connection.go
@@ -198,7 +198,7 @@ func (s *SshConnection) connect() error {
 
 	identityPath := s.identity
 	if s.identity == "" {
-		usr:= utils.CurrentUser()
+		usr := utils.CurrentUser()
 		identityPath = filepath.Join(usr.HomeDir, ".ssh", "id_rsa")
 	}
 

--- a/pkg/sshc/ssh_connection.go
+++ b/pkg/sshc/ssh_connection.go
@@ -6,7 +6,6 @@ import (
 	"io"
 	"net"
 	"os"
-	"os/user"
 	"path/filepath"
 	"sync"
 	"sync/atomic"
@@ -63,7 +62,7 @@ func NewSshConnection(conf *SshClientConf) *SshConnection {
 	parsed := utils.ParseSSHUrl(conf.ServerURI)
 	var knownHostsPath string
 	if conf.KnownHosts == "" {
-		usr, _ := user.Current()
+		usr := utils.CurrentUser()
 		knownHostsPath = filepath.Join(usr.HomeDir, ".ssh", "known_hosts")
 	} else {
 		knownHostsPath, _ = utils.ExpandUserHome(conf.KnownHosts)
@@ -199,7 +198,7 @@ func (s *SshConnection) connect() error {
 
 	identityPath := s.identity
 	if s.identity == "" {
-		usr, _ := user.Current()
+		usr:= utils.CurrentUser()
 		identityPath = filepath.Join(usr.HomeDir, ".ssh", "id_rsa")
 	}
 

--- a/pkg/sshd/channel_handler.go
+++ b/pkg/sshd/channel_handler.go
@@ -7,7 +7,6 @@ import (
 	"net"
 	"os"
 	"os/exec"
-	"os/user"
 	"runtime"
 	"strings"
 	"sync"
@@ -56,10 +55,7 @@ func (s *channelHandler) handleShellExecRequest(
 	var shell string
 
 	if s.server.shellExecutable == "" {
-		usr, err := user.Current()
-		if err != nil {
-			panic(err)
-		}
+		usr := utils.CurrentUser()
 		shell = utils.GetUserDefaultShell(usr.Username)
 	} else {
 		shell = s.server.shellExecutable
@@ -96,7 +92,7 @@ func (s *channelHandler) handleShellExecRequest(
 		envVal = append(envVal, fmt.Sprintf("%s=%s", k, v))
 	}
 
-	usr, _ := user.Current()
+	usr := utils.CurrentUser()
 
 	// export TERM
 	term := os.Getenv("TERM")

--- a/pkg/utils/helpers.go
+++ b/pkg/utils/helpers.go
@@ -168,8 +168,8 @@ func CurrentUser() *user.User {
 				return "/"
 			}()
 			u := &user.User{
-				Uid:      "0",
-				Gid:      "0",
+				Uid:      "?",
+				Gid:      "?",
 				Username: userName,
 				Name:     userName,
 				HomeDir:  homeDir,

--- a/pkg/utils/helpers.go
+++ b/pkg/utils/helpers.go
@@ -11,6 +11,7 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"sync"
 )
 
 const (
@@ -28,7 +29,7 @@ type sshUrl struct {
 func ParseSSHUrl(url string) *sshUrl {
 	parts := strings.Split(url, "@")
 
-	usr, _ := user.Current()
+	usr := CurrentUser()
 	conf := &sshUrl{}
 
 	var hostPort string
@@ -76,10 +77,7 @@ func ParseSSHUrl(url string) *sshUrl {
 
 // ExpandUserHome resolve paths like "~/.ssh/id_rsa"
 func ExpandUserHome(path string) (string, error) {
-	usr, err := user.Current()
-	if err != nil {
-		return "", err
-	}
+	usr := CurrentUser()
 	ret := path
 
 	// supports paths like "~/.ssh/id_rsa"
@@ -134,4 +132,50 @@ func ByteCountSI(b int64) string {
 	}
 	return fmt.Sprintf("%.1f %cB",
 		float64(b)/float64(div), "kMGTPE"[exp])
+}
+
+var currentUserCache struct {
+	sync.Once
+	u *user.User
+}
+
+func CurrentUser() *user.User {
+	currentUserCache.Do(func() {
+		currentUserCache.u = func() *user.User {
+			if cu, err := user.Current(); err == nil {
+				return cu
+			}
+			// error fallback
+			userName := func() string {
+				if runtime.GOOS == "windows" {
+					if user, ok := os.LookupEnv("USERNAME"); ok {
+						return user
+					}
+				} else {
+					if user, ok := os.LookupEnv("USER"); ok {
+						return user
+					}
+				}
+				return "root"
+			}()
+			homeDir := func() string {
+				if home, err := os.UserHomeDir(); err == nil {
+					return home
+				}
+				if wd, err := os.Getwd(); err == nil {
+					return wd
+				}
+				return "/"
+			}()
+			u := &user.User{
+				Uid:      "0",
+				Gid:      "0",
+				Username: userName,
+				Name:     userName,
+				HomeDir:  homeDir,
+			}
+			return u
+		}()
+	})
+	return currentUserCache.u
 }

--- a/pkg/utils/keys.go
+++ b/pkg/utils/keys.go
@@ -11,7 +11,6 @@ import (
 	"log"
 	"net"
 	"os"
-	"os/user"
 	"path/filepath"
 
 	"golang.org/x/crypto/ssh"
@@ -69,7 +68,7 @@ func WriteKeyToFile(keyBytes []byte, keyPath string) error {
 func LoadIdentityFile(file string) (ssh.AuthMethod, error) {
 	path, _ := ExpandUserHome(file)
 
-	usr, _ := user.Current()
+	usr := CurrentUser()
 	// no path is set, try with a reasonable default
 	if path == "" {
 		path = filepath.Join(usr.HomeDir, ".ssh", "id_rsa")

